src/stream.ts

```diff
@@ -160,11 +155,37 @@ export class WebRTCStream implements Stream {
       if (closed || this.writeClosed) {
         break;
       }
-      let res = buf.subarray();
-      let send_buf = pb.Message.toBinary({ message: buf.subarray() });
-      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${send_buf}`);
-      this.channel.send(send_buf);
+      const res = buf.subarray();
+      const msgbuf = pb.Message.toBinary({message: buf.subarray()});
+      const sendbuf = lp.encode.single(msgbuf)
+      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${msgbuf}`);
+      this.channel.send(sendbuf.subarray())
+    }
+  }
+
+  processIncomingProtobuf(buffer: Uint8Array): { data: Uint8Array | undefined } {
```

Comment:
why this weird return structure? e.g. how come not just `Uint8Array | undefined`, what's the point of wrapping it in a struct, if you are always unpacking it anyway?!